src/app/page/Recipient/RequestsList.tsx

```diff
@@ -0,0 +1,227 @@
+    width: "100%",
+    justifyContent: "center",
+    alignItems: "center",
+    "& .MuiAvatar-root": {
+      margin: "0 .5rem",
+    },
+  },
+}));
+
+type Props = {
+  missions: MissionInterface[];
+  fallback: string;
+};
+
+export default function ({ fallback, missions }: Props) {
+  const classes = useStyles();
+
+  if (missions.length < 1) {
+    return (
+      <Typography className={classes.fallback} align="center" variant="body1" color="textSecondary">
+        {fallback}
+      </Typography>
+    );
+  }
+
+  return (
+    <>
+      {missions.map((mission) => (
+        <RecipientMissionCard key={mission.uid} mission={mission} />
+      ))}
+    </>
+  );
+}
+
+function RecipientMissionCard({ mission }: { mission: MissionInterface }) {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  const toggle = () => setOpen(!open);
+
+  return (
+    <Card className={classes.card}>
+      <TopBar
+        status={mission.status}
+        fundedStatus={mission.fundedStatus}
+        createdDate={mission.createdDate}
+      />
+      <CardActionArea onClick={toggle}>
+        <CardHeader
+          className={classes.cardHeader}
+          avatar={<AppleIcon className={classes.appleIcon} />}
+          title={<Typography variant="h4">FOOD BOX DELIVERY</Typography>}
+          subheader={mission.details?.map((r) => (
+            <Typography key={r.resourceUid} variant="subtitle2">
+              {r.quantity} x {r.displayName}
+            </Typography>
+          ))}
+        />
+      </CardActionArea>
+      <Collapse in={open}>
+        <Details mission={mission} />
+        {mission.status === Status.delivered && <ConfirmDelivery mission={mission} />}
+      </Collapse>
+      <CardActionArea onClick={toggle} className={classes.cardBottomToggle}>
+        <Typography align="center" color="primary">
+          {open ? "HIDE DETAILS" : "MORE DETAILS"}
+        </Typography>
+      </CardActionArea>
+    </Card>
+  );
+}
+
+type TopBarProps = {
+  status: Status;
+  fundedStatus: FundedStatus;
+  createdDate: string;
+  className?: string;
+};
+
+function Icon({ fundedStatus, status, ...rest }: any) {
+  if (status === Status.started) return <PlayCircleFilled {...rest} />;
+  if (status === Status.delivered) return <NotificationImportant {...rest} />;
+  if (status === Status.succeeded) return <CheckCircle {...rest} />;
+  if (fundedStatus === FundedStatus.notfunded) return <HourglassFull {...rest} />;
+  return <FiberNew {...rest} />;
+}
+
+function TopBarText({ createdDate, fundedStatus, status }: any) {
+  const date = new Date(createdDate).toLocaleDateString();
+
+  if (status === Status.started) return <>In progress | Accepted</>;
+  if (status === Status.delivered) return <>It's here! Confirm delivery below</>;
+  if (status === Status.succeeded) return <>Delivery Confirmed</>;
+  if (fundedStatus === FundedStatus.notfunded) return <>Awaiting donation | Submitted {date}</>;
+  return <>Submitted {date}</>;
```

Comment:
Lol Hmm i mean it... works.. It's actually much easier to read with this many if's. I just feel like this can be a littler DRY-er somehow... But i think we're good for now. 

Does this need to return text in a fragment though? Should that just be a string?

Reply:
I was tempted to share the Icon and text state in the same component as they share the same return states, but that isn't necessarily guaranteed going forward as the designs don't really show all the possible combinations that the top bar can be in. Keeping it separate should make future changes easier if/when these states diverge.

I had initially used fragment as typescript has an issue since `string` isn't a jsx constructor. Apparently still an ongoing issue (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20544)

Went back and was able to cheat it with `any` though so guess that works lol